src/modules/users/controllers/users.js

```diff
@@ -0,0 +1,86 @@
+const bcrypt = require('bcrypt')
+
+const User = require('../models/User')
+const usernameGenerator = require('../utils/usernameGenerator')
+
+const userModel = User()
+
+const create = async (req, res) => {
+  const { nome, senha, idPapel } = req.body
+
+  try {
+    const username = usernameGenerator.createUsernameFromName(nome)
+
+    const hashPassword = await bcrypt.hash(senha, 10)
+
+    const userId = await userModel.create({ nome: username, senha: hashPassword, idPapel })
+
+    res.status(200).send({ id: userId, nome: username })
```

Comment:
```suggestion
    const userId = await userModel.create({ username, senha: hashPassword, idPapel, nome })

    res.status(200).send({ id: userId, username, nome })
```
Acredito que vale salvar o nome, vamos exibir o nome do usuário no front, vai ficar estranho usar "admin3521" por exemplo.

Reply:
Certo, então farei a alteração na tabela também, pois substituí o name por username no banco